volttron/platform/control.py

```diff
@@ -840,6 +840,131 @@ def list_agent_rpc_code(opts):
+        user id of a ZMQ credential, or the common name of a CSR
+        remote address of the credential or csr
+        status of the credential or cert (either APPROVED, DENIED, or PENDING)
+
+     """
+    conn = opts.connection
+    if not conn:
+        _stderr.write("VOLTTRON is not running. This command "
+                      "requires VOLTTRON platform to be running\n")
+        return
+
+    output_view = []
+    try:
+        pending_csrs = conn.server.vip.rpc.call(AUTH, "get_pending_csrs").get(timeout=4)
+        for csr in pending_csrs:
+            output_view.append({"entry": {"user_id": csr["identity"],
+                                          "address": csr["remote_ip_address"]},
+                                "status": csr["status"]
+                                })
+    except TimeoutError:
+        print("Certs timed out")
+    try:
+        approved_certs = conn.server.vip.rpc.call(AUTH, "get_authorization_approved").get(timeout=4)
+        for value in approved_certs:
+            output_view.append({"entry": value, "status": "APPROVED"})
+    except TimeoutError:
+        print("Approved credentials timed out")
+    try:
+        denied_certs = conn.server.vip.rpc.call(AUTH, "get_authorization_denied").get(timeout=4)
+        for value in denied_certs:
+            output_view.append({"entry": value, "status": "DENIED"})
+    except TimeoutError:
+        print("Denied credentials timed out")
+    try:
+        pending_certs = conn.server.vip.rpc.call(AUTH, "get_authorization_failures").get(timeout=4)
+        for value in pending_certs:
+            output_view.append({"entry": value, "status": "PENDING"})
+    except TimeoutError:
+        print("Pending credentials timed out")
+
+    if len(output_view) == 0:
+        print("No remote certificates or credentials")
+        return
+
+    if opts.status == "approved":
+        output_view = [output for output in output_view if output["status"] == "APPROVED"]
+
+    elif opts.status == "denied":
+        output_view = [output for output in output_view if output["status"] == "DENIED"]
+
+    elif opts.status == "pending":
+        output_view = [output for output in output_view if output["status"] == "PENDING"]
+
+    elif opts.status is not None:
+        _stdout.write("Invalid parameter. Please use 'approved', 'denied', 'pending', or leave blank to list all.\n")
+        return
+
+    if len(output_view) == 0:
+        print(f"No {opts.status} remote certificates or credentials")
+        return
+
+    for output in output_view:
+        for value in output["entry"]:
+            if not output["entry"][value]:
+                output["entry"][value] = "-"
+
+    userid_width = max(5, max(len(str(output["entry"]["user_id"])) for output in output_view))
+    address_width = max(5, max(len(str(output["entry"]["address"])) for output in output_view))
+    status_width = max(5, max(len(str(output["status"])) for output in output_view))
+    fmt = '{:{}} {:{}} {:{}}\n'
```

Comment:
Not a big deal, but assuming that _stderr.write() takes a string, have you tried using [f-strings](https://docs.python.org/3/whatsnew/3.6.html) for the error message? Doing so would make it read (and be more readable) as such:

```python
_stderr.write(f"USER_ID: {userid_width}, ADDRESS: {address_width}, STATUS: {status_width}")
```

Reply:
Keeping it the same at this time to match format for agent status output. Worth noting.